### PR TITLE
Improve slap of Find Command and relevant files 

### LINKED ISSUE #278: Improve slap of Find Command and relevant files
As @Zer0Legion has pointed out, the find command can be improved further to incorporate slap and reduce the number of layers in the code.

### PR DESCRIPTION
**Pull request by jayllo-c**
Fixes #278 

Currently, the find command's functionality has no issue. However as
@Zer0Legion pointed out, the functions can be further abstracted.

We need to make the changes as he has suggested to improve the
readability of the code in general.

Let's incorporate SLAP into some nested functions to reduce the amount
of nested code.

In doing so, we can promote code clarity, maintainability, and
reusability, where functions and modules are focused on completing one
task at a time.

Note that no amends to the test code are required since the abstraction is 
done in such a way that the code coverage for these files is still 100%.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -59,20 +59,30 @@ public CommandResult execute(Model model) throws CommandException {
...
                 String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
     }
 
+    /**
+     * Updates the filtered person list by score.
+     * @param model
+     * @param selectedExam
+     * @throws CommandException
+     */
+    private void updateFilteredPersonListByScore(Model model, Exam selectedExam) throws CommandException {
```
Comment (delishad21) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:76]: good slapping

**Review by delishad21 — COMMENTED** [review]
Simple code refactoring but makes the code a lot more readable

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 1 above)
Comment (Pughal77) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:76]: Great abstraction here

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -59,20 +59,30 @@ public CommandResult execute(Model model) throws CommandException {
...
-            model.updateFilteredPersonList(predicate.withExam(selectedExam));
+            updateFilteredPersonListByScore(model, selectedExam);
         } else {
             model.updateFilteredPersonList(predicate);
         }
         return new CommandResult(
                 String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
     }
 
+    /**
```
Comment (danielleloh) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:70]: good catch to improve slap of code :)

**Review by Pughal77 — COMMENTED** [review]
Looks Good to me!

**Pull request merged**